certification/internal/shell/validate_operator_bundle.go

```diff
@@ -1,42 +1,59 @@
 package shell
 
 import (
-	"os/exec"
-	"strings"
-
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
-type ValidateOperatorBundlePolicy struct {
+type ValidateOperatorBundleCheck struct {
 }
 
-func (p ValidateOperatorBundlePolicy) Validate(bundle string) (bool, error) {
-	stdouterr, err := exec.Command("operator-sdk", "bundle", "validate", "-b", "podman", "--verbose", bundle).CombinedOutput()
+func (p ValidateOperatorBundleCheck) Validate(bundle string) (bool, error) {
+	report, err := p.getDataToValidate(bundle)
 	if err != nil {
 		log.Error("Error will executing operator-sdk validate bundle: ", err)
 		return false, err
 	}
 
-	lines := strings.Split(string(stdouterr), "time=")
+	return p.validate(report)
+}
+
+func (p ValidateOperatorBundleCheck) getDataToValidate(bundle string) (*cli.OperatorSdkBundleValidateReport, error) {
+	selector := []string{"community", "operatorhub"}
+	opts := cli.OperatorSdkBundleValidateOptions{
+		Selector:        selector,
+		Verbose:         true,
+		ContainerEngine: "podman",
+		OutputFormat:    "json-alpha1",
+	}
+
+	return operatorSdkEngine.BundleValidate(bundle, opts)
+}
 
-	if strings.Contains(lines[len(lines)-1], "All validation tests have completed successfully") {
-		for _, line := range lines {
-			if strings.Contains(line, "level=warning") {
-				log.Warn("time= ", line)
+func (p ValidateOperatorBundleCheck) validate(report *cli.OperatorSdkBundleValidateReport) (bool, error) {
+	if !report.Passed || len(report.Outputs) > 0 {
+		for _, output := range report.Outputs {
+			var logFn func(...interface{})
+			switch output.Type {
+			case "warning":
+				logFn = log.Warn
+			case "error":
+				logFn = log.Error
+			default:
+				logFn = log.Debug
 			}
+			logFn(output.Message)
 		}
```

Comment:
This is taking the output directly from `operator-sdk bundle validate` and mapping to our logging facilities in preflight, correct? This is a tricky approach for me; it directly ties the logging structure of operator-sdk to our logging structure. If their logging structure changes, we're unlikely to see it prior to shipping.

I know it's probably less pretty, but from the above perspective, would it be more stable to just pretty-print the entire log output at logging verbosity that we control so that the user has visibility into why something may have triggered a failure case? 

I do think we have this on other places as well. I'm not sure we need to solve it in this PR, but I did want to mention it to see what we could to give the user the needed visibility into things without hooking too deeply into logging behaviors which may not be reliable for our use case.

Also, if I misunderstood what this is doing, let me know/resolve away!

Reply:
It really it just for reporting. We are in no way relying on this output to validate the check. I just wanted to log the messages at the appropriate level. It could just be log.Info for any of them.

Reply:
Definitely agree that the reporting is necessary. Just wary about associating the operator-sdk log levels to ours. I think I would prefer the entire log output from invoked tool just be a single log level on our side (some higher log level). 

Starting to feel like a nit, so I think we can table this and disregard. I'll leave it to your discretion 👍

Reply:
As it isn't there log levels, but actual json output from operator-sdk, I'm ok with this one for the time being.